juspay/hyperswitch
# fix: return 400 https status code for authentication failure

## Type of Change



## Description

Currently we are returning 200 status code if authentication call fails at authetication connector.

Made some changes to return 400 in case authentication call fails at authetication connector.


### Additional Changes





## Motivation and Context



## How did you test it?

Manual.
<img width="1728" alt="Screenshot 2024-04-18 at 12 28 13 PM" src="https://github.com/juspay/hyperswitch/assets/61539176/2553b359-4c31-4071-89b9-42000c282982">

1. Create a Merchant Account with threedsecure.io and netcetera.
2. Create a payment
```
curl --location 'https://sandbox.hyperswitch.io/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_PSTUyULJHdsuODU0JcEjuXDJdXkeEQXc6Ef2f3dFx4vlST9bKjmMjxfq4DVXppeE' \
--data-raw '{
    "amount": 6540,
    "currency": "USD",
    "confirm": false,
    "capture_method": "automatic",
    "capture_on": "2022-09-10T10:11:12Z",
    "amount_to_capture": 6540,
    "customer_id": "StripeCustomer",
    "email": "guest@example.com",
    "name": "John Doe",
    "phone": "999999999",
    "phone_country_code": "+65",
    "description": "Its my first payment request",
    "authentication_type": "three_ds",
    "return_url": "https://duck.com",
    "profile_id": "pro_iHTSgIRARnO8Iqx0HGcN",
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "PiX"
        }
    },
    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "PiX"
        }
    },
    "request_external_three_ds_authentication": true,
    "statement_descriptor_name": "joseph",
    "statement_descriptor_suffix": "JS",
    "metadata": {
        "udf1": "value1",
        "new_customer": "true",
        "login_date": "2019-09-10T10:11:12Z"
    }
}'
```
3. Confirm the payment
```
curl --location 'https://sandbox.hyperswitch.io/payments/pay_yosQqaFWdtjwoTpSAwoB/confirm' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: pk_snd_23ff7c6d50e5424ba2e88415772380cd' \
--header 'Cookie: login_token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoiMmY2YzVlOTEtYzdkMy00NzI5LWI0YjMtMTdiNGY2YjExYWQwIiwibWVyY2hhbnRfaWQiOiJtZXJjaGFudF8xNzAxMTY4Njk5Iiwicm9sZV9pZCI6Im9yZ19hZG1pbiIsImV4cCI6MTcxNTQyMDAzOCwib3JnX2lkIjoib3JnX2ppQmRyVVJqT2Nndk5mQzhDVlFwIn0.JtK5iUvfhkPkqJ1TJhRFwb3O1mzKWF4FCgHcQ2K7ZLw' \
--data '{
    "browser_info": {
        "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36",
        "accept_header": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8",
        "language": "nl-NL",
        "color_depth": 24,
        "screen_height": 723,
        "screen_width": 1536,
        "time_zone": 0,
        "java_enabled": true,
        "java_script_enabled": true,
        "ip_address": "115.99.183.2"
    },
    "client_secret": "pay_yosQqaFWdtjwoTpSAwoB_secret_LXATHDTt22mnKUr6Aw4t",
    "payment_method": "card",
    "payment_method_data": {
        "card": {
            "card_number": "4000100511112003",
            "card_exp_month": "10",
            "card_exp_year": "25",
            "card_holder_name": "joseph Doe",
            "card_cvc": "123"
        }
    }
}
'
```
4. Do authentication call without completing 3ds method collection. Authentication will fail.(Should get 400 status code)
```
curl --location 'https://sandbox.hyperswitch.io/payments/pay_yosQqaFWdtjwoTpSAwoB/3ds/authentication' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: pk_snd_23ff7c6d50e5424ba2e88415772380cd' \
--data '{
    "client_secret": "pay_yosQqaFWdtjwoTpSAwoB_secret_LXATHDTt22mnKUr6Aw4t",
    "device_channel": "BRW",
    "threeds_method_comp_ind": "Y"
}'
```

## Checklist
